File: library/src/main/java/me/shkschneider/skeleton/java/ArrayHelper.java
package me.shkschneider.skeleton.java;

import android.support.annotation.IntRange;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.Arrays;
import java.util.List;

public class ArrayHelper {

    protected ArrayHelper() {
        // Empty
    }

    @IntRange(from=-1)
    public static <T> int index(@NonNull final T[] objects, @NonNull final T object) {
        for (int i = 0; i < objects.length; i++) {
            if (objects[i].equals(object)) {
                return i;
            }
        }
        return -1;
    }

    public static <T> List<T> list(@NonNull final T[] objects) {
        return Arrays.asList(objects);
    }

    public static <T> boolean contains(@NonNull final T[] objects, @NonNull final T object) {
        return ListHelper.contains(list(objects), object);
    }

    @Nullable
    public static <T> T first(@NonNull final T[] objects) {
        return ListHelper.first(list(objects));
    }

    @Nullable
    public static <T> T last(@NonNull final T[] objects) {
        return ListHelper.last(list(objects));
    }

}

File: library/src/main/kotlin/me/shkschneider/skeleton/java/ArrayHelper.kt
package me.shkschneider.skeleton.java

import android.support.annotation.IntRange

import java.util.Arrays

object ArrayHelper {

    @IntRange(from = -1)
    fun <T> index(objects: Array<T>, obj: T): Int {
        return objects.indices.firstOrNull { objects[it] == obj } ?: -1
    }

    fun <T> list(objects: Array<T>): List<T> {
        return Arrays.asList(*objects)
    }

    fun <T> contains(objects: Array<T>, obj: T): Boolean {
        return ListHelper.contains(list(objects), obj)
    }

    fun <T> first(objects: Array<T>): T? {
        return ListHelper.first(list(objects))
    }

    fun <T> last(objects: Array<T>): T? {
        return ListHelper.last(list(objects))
    }

}
